### PR TITLE
TypeScript config

### PR DESCRIPTION
**Pull request by visormatt**
> I saw that you published the **1.0**, I'm sipping scotch to celebrate! 

You had me asking some questions at the end of it all and I think I've figured a bit more out as well. 

First, If you'd still like to make `Time` available for the consumer we can simply export it from the `index` file. As we're creating an NPM 📦 we only point to a single file **/dist/index.js**. It can and will make use of other imports, but anything we want exposed to the `consumer` has to be explicitly exported 💡 Think of that file as the `access point` which is set in the `package.json` as `main ~ or ~ module`.

Next, your use of `.d.ts` was perfectly acceptable, I think I've just operated under a different mental model. Really there are two sets of `type definitions`. There is the source code which in this case is TypeScript. You can totally hand carve your own definitions as you had it, I think I've just always thought of it as the source 🤷‍♂️ The **second set** though, is the type definitions that are generated from the code. These too could be manually create but you're already automating it. These definitions are what bridge the gap from the `dist` folder (consumer code) which is now `vanilla js` and thus has no concept of types. 
> I renamed `time.ts -> time.d.ts` and moved it to the `src` which works (kinda)... However when running the build, any `.d.ts` files that are created manually aren't also copied to the types destination. So, you can do it, but then you'll have to have it in the same folder we're outputting the generated types which feels a bit dirty to merge the two. Personally I'd stick with treating it as source code vs. a definition and let the compiler generate all the type definitions. 

Lastly, if you wanted to make it **all** available I've seen and read about folks just including the `src` directory. Then if a consumer is already working in TypeScript they can bundle it any way they see fit and access that `sugar` that typescript offers us without having to manually export it all from our access point. 

## Testing

- [ ] Should still build as expected
- [ ] Types are unaffected in the example (still working)
- [ ] Time should be importable from the package (see example)

Moving the `types` folder and killing `rootDir` were how I initially understood you're types folder, but I think this still mostly fits, however the folder is used for `generated types` vs. the `manual types` which the extension has been changed back to `.d.ts`.

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -43,4 +43,4 @@ const useCountdown = ({minutes = 0, seconds = 0}: useCountdownParams): Time => {
     return countdown
 }
 
-export default useCountdown
+export {useCountdown as default, Time, useCountdown}
```
Comment (visormatt) [inline src/index.tsx]: Left this as default so it's non-breaking in addition to now making Time and useCountdown available as named exports.

File: .gitignore [hunk 2]
```diff
@@ -7,6 +7,7 @@ node_modules
 
 # build
 dist
+types
```
Comment (visormatt) [inline .gitignore]: As I noted I pulled these up as an example but feel free to use it in `dist/types` if you liked that config.

File: tsconfig.json [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
+    "include": ["src"],
     "compilerOptions": {
         "module": "esnext",
         "lib": ["dom", "esnext"],
         "importHelpers": true,
         "declaration": true,
-        "declarationDir": "dist/types",
+        "declarationDir": "types",
         "sourceMap": true,
-        "rootDir": "./src",
```
Comment (visormatt) [inline tsconfig.json:10]: Killed this as you've already scoped it in the `include` at the top, making the root the same as the location of this `tsconfig` file.

File: example/index.tsx [hunk 4]
```diff
@@ -3,10 +3,12 @@ import "react-app-polyfill/ie11"
 import * as React from "react"
 import * as ReactDOM from "react-dom"
 
-import useCountdown from "../."
+import useCountdown, {Time} from "../."
```
Comment (visormatt) [inline example/index.tsx:6]: 🎉 now you can import it from the package directly

File: package.json [hunk 5]
```diff
@@ -27,7 +27,7 @@
     ],
     "main": "dist/index.js",
     "module": "dist/use-countdown.esm.js",
-    "types": "dist/types/index.d.ts",
+    "types": "types/index.d.ts",
```
Comment (visormatt) [inline package.json]: Related to the hoisting of types directory

File: tsconfig.json [hunk 6]
```diff
@@ -1,20 +1,20 @@
...
         "strict": true,
         "noImplicitReturns": true,
         "noFallthroughCasesInSwitch": true,
         "noUnusedLocals": true,
         "noUnusedParameters": true,
         "moduleResolution": "node",
         "jsx": "react",
+        "removeComments": false,
```
Comment (visormatt) [inline tsconfig.json:17]: This is specifically the TS comments which will show in your `.d.ts`, the build process should remove any comments from the actual `dist` code.

File: example/index.tsx (hunk 4 above)
Comment (bradgarropy) [reply example/index.tsx:6]: Love this!

File: .gitignore (hunk 2 above)
Comment (bradgarropy) [reply .gitignore]: I do prefer them to be generated at `dist/types`.

File: example/index.tsx [hunk 7]
```diff
@@ -3,10 +3,12 @@ import "react-app-polyfill/ie11"
...
 
 const App = () => {
+    const _time: Time = {minutes: 0, seconds: 0}
```
Comment (bradgarropy) [inline example/index.tsx]: Yep, awesome!

File: tsconfig.json (hunk 6 above)
Comment (bradgarropy) [reply tsconfig.json:17]: Thank you for this one!

**Review by bradgarropy — APPROVED** [review]
Thanks for the PR and the awesome explanation/documentation! I can tell you're a seasoned vet at this kind of stuff.

I pushed some changes directly to this branch, mainly keeping the generated types at `dist/types`. So I'll go ahead and approve!

Thank you again!!!

**Comment by visormatt** [comment]
Sweet 🎉 I enjoyed having to dig into things a bit myself. **ABL** (always be learning) in our world, keep up the great work!

**Comment by bradgarropy** [comment]
@all-contributors add @visormatt for code and docs

**Pull request merged**